File: SunnyFlamingo/Models/QuestionBase.cs
using SunnyFlamingo.ValueObjects;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SunnyFlamingo.Models
{
    public class QuestionBase<T>
    {
        public string QuestionsKey { get; set; }
        public T Value { get; set; }
        public string Key { get; set; }
        public string Label { get; set; }
        public bool Required { get; set; }
        public int? Order { get; set; }
        public ControlType ControlType { get; set; }
        public InputType Type { get; set; }
        public List<QuestionOption> Options { get; set; }
        public bool? Checked { get; set; }
        public int? AfterBox2 { get; set; }
        public string AfterBox { get; set; }
        public object SelectedFrom { get; set; }
        public object SelectedTo { get; set; }
        public object From { get; set; }
        public object To { get; set; }
    }
}

File: SunnyFlamingo/ClientApp/src/app/models/question-base.ts
import { ControlType } from "../value-objects/control-type";
import { InputType } from "../value-objects/input-type";
import { QuestionOption } from "./question-option";
import { QuestionValidator } from "./question-validator";

export class QuestionBase<T> {
  value?: T;
  key: string;
  label: string;
  order: number;
  controlType?: ControlType;
  type?: InputType;
  options: QuestionOption[];
  checked?: boolean;
  afterBox?: number;
  selectedFrom?: number;
  selectedTo?: number;
  from?: number;
  to?: number;
  validator?: QuestionValidator;
  methodName: string;

  constructor(options: {
      value?: T;
      key?: string;
      label?: string;
      order?: number;
      controlType?: ControlType;
      type?: InputType;
      options?: QuestionOption[];
      checked?: boolean;
      afterBox?: number;
      selectedFrom?: number;
      selectedTo?: number;
      from?: number;
      to?: number;
      validator?: QuestionValidator;
      methodName?: string;
    } = {}) {
    this.value = options.value;
    this.key = options.key || '';
    this.label = options.label || '';
    this.order = options.order === undefined ? 1 : options.order;
    this.controlType = options.controlType;
    this.type = options.type;
    this.options = options.options || [];
    this.checked = options.checked;
    this.afterBox = options.afterBox;
    this.selectedFrom = options.selectedFrom;
    this.selectedTo = options.selectedTo;
    this.from = options.from;
    this.to = options.to;
    this.validator = options.validator;
    this.methodName = options.methodName || '';
  }
}